nrf-hal-common/src/ieee802154.rs

```diff
@@ -326,6 +326,29 @@ impl<'c> Radio<'c> {
         }
     }
 
+    pub fn recv_async_start(&mut self, packet: &mut Packet) -> () {
+        unsafe {
+            self.start_recv(packet);
```

Comment:
This is unsound. You're starting DMA into `packet` and returning, but it is only borrowed for the duration of this function. The user can deallocate or reuse the `packet` memory for something else, and DMA will write all over it causing memory corruption.

Reply:
I am fully aware of this fact, but not familiar enough with rust yet as to know how to implement it properly.
Pointers / link to code doing something similar the proper way would be greatly appreciated.